internal/service/verify/resource_verify_policy_test.go

```diff
@@ -0,0 +1,657 @@
+		return err
+	}
+
+	apiClient := p1Client.API.VerifyAPIClient
+	ctx = context.WithValue(ctx, verify.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	mgmtApiClient := p1Client.API.ManagementAPIClient
+	ctx = context.WithValue(ctx, management.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "pingone_verify_policy" {
+			continue
+		}
+
+		_, rEnv, err := mgmtApiClient.EnvironmentsApi.ReadOneEnvironment(ctx, rs.Primary.Attributes["environment_id"]).Execute()
+
+		if err != nil {
+
+			if rEnv == nil {
+				return fmt.Errorf("Response object does not exist and no error detected")
+			}
+
+			if rEnv.StatusCode == 404 {
+				continue
+			}
+
+			return err
+		}
+
+		body, r, err := apiClient.VerifyPoliciesApi.ReadOneVerifyPolicy(ctx, rs.Primary.Attributes["environment_id"], rs.Primary.Attributes["id"]).Execute()
+
+		if err != nil {
+
+			if r == nil {
+				return fmt.Errorf("Response object does not exist and no error detected")
+			}
+
+			if r.StatusCode == 404 {
+				continue
+			}
+
+			tflog.Error(ctx, fmt.Sprintf("Error: %v", body))
+			return err
+		}
+
+		return fmt.Errorf("PingOne Verify Policy %s still exists", rs.Primary.ID)
+	}
+
+	return nil
+}
+
+func TestAccVerifyPolicy_NewEnv(t *testing.T) {
+	t.Parallel()
+
+	resourceName := acctest.ResourceNameGen()
+	resourceFullName := fmt.Sprintf("pingone_verify_policy.%s", resourceName)
+
+	environmentName := acctest.ResourceNameGenEnvironment()
+
+	name := resourceName
+
+	licenseID := os.Getenv("PINGONE_LICENSE_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheckEnvironment(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckVerifyPolicyDestroy,
+		ErrorCheck:               acctest.ErrorCheck(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVerifyPolicyConfig_NewEnv(environmentName, licenseID, resourceName, name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(resourceFullName, "id", validation.P1ResourceIDRegexp),
+				),
+			},
+		},
+	})
+}
+
+func TestAccVerifyPolicy_Full(t *testing.T) {
+	t.Parallel()
+
+	resourceName := acctest.ResourceNameGen()
+	resourceFullName := fmt.Sprintf("pingone_verify_policy.%s", resourceName)
+
+	name := acctest.ResourceNameGen()
+	updatedName := acctest.ResourceNameGen()
+
+	environmentName := acctest.ResourceNameGenEnvironment()
+	licenseID := os.Getenv("PINGONE_LICENSE_ID")
+
+	fullPolicy := resource.ComposeTestCheckFunc(
+		resource.TestMatchResourceAttr(resourceFullName, "id", validation.P1ResourceIDRegexp),
+		resource.TestMatchResourceAttr(resourceFullName, "environment_id", validation.P1ResourceIDRegexp),
+		resource.TestCheckResourceAttr(resourceFullName, "name", name),
+		resource.TestCheckResourceAttr(resourceFullName, "description", name),
```

Comment:
Apologies this wasn't picked up on previous review - this test on the `description` field should be a unique value relative to other attributes, this catches any copy-paste where (for example) `name` in config/plan is accidentally mapped to `description` in state

Reply:
resolved